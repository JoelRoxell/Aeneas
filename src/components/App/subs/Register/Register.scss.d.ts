// This file is generated automatically
export const center: string;
export const register: string;
export const view: string;
export const title: string;
