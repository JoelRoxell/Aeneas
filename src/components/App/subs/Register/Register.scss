.register {
  display: flex;
  justify-content: center;
  flex: 1;

  .view {
    display: flex;
    flex-direction: column;
    margin-top: rem(50);
    background: white;
    width: rem(457);
    height: fit-content;
    padding: rem(59) rem(90) rem(65) rem(90);
    box-shadow: 0 rem(2) rem(21) 0 rgba(0, 0, 0, 0.08);

    .title {
      color: black;
      text-align: center;
    }
  }
}
