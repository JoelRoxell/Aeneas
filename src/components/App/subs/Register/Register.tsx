import * as React from 'react'
import { connect } from 'react-redux'
import { Switch, Route } from 'react-router'

import * as style from './Register.scss'
import { ApplicationState, ConnectedReduxProps } from 'store'

interface RegisterProps extends ConnectedReduxProps {
  loading: boolean
  email: string
  name: string
  contact: string
  password: string
  agreedTerms: boolean
  error: string
  token: string
}

class Register extends React.Component<RegisterProps> {
  render() {
    return (
      <div className={style.register}>
        <div className={style.view}>
          <Switch>
            <Route
              path="/register/podcaster"
              render={() => {
                return <div />
              }}
            />
          </Switch>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state: ApplicationState) => {
  return {
    loading: state.auth.loading,
    email: state.auth.email,
    name: state.auth.name,
    contact: state.auth.contact,
    password: state.auth.password,
    agreedTerms: state.auth.agreedTerms,
    error: state.auth.error,
    token: state.auth.token
  }
}

export default connect(mapStateToProps)(Register)
