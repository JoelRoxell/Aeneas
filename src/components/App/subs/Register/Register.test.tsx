import * as React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'

import Register from 'components/App/subs/Register'
import UserProvider from 'contexts/UserContext'

describe('Register', () => {
  test('render', () => {
    const wrapper = mount(
      <UserProvider>
        <Router>
          <Register />
        </Router>
      </UserProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
