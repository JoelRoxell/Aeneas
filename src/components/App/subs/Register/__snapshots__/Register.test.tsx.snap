// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register render 1`] = `
<UserProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Register>
        <div
          className="register"
        >
          <div
            className="view"
          >
            <form>
              <Input
                autoComplete="email"
                name="email"
                onChange={[Function]}
                placeholder="email"
                rightIcon="user"
                title="User"
                validate={[Function]}
                value=""
              >
                <div
                  className="input"
                >
                  <div
                    className="label"
                  >
                    User
                  </div>
                  <div
                    className="inputWrapper"
                  >
                    <input
                      autoComplete="email"
                      className="textField"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="email"
                      type="text"
                      value=""
                    />
                    <Icon
                      name="user"
                      svgStyle={
                        Object {
                          "className": "rightIcon",
                          "height": 24,
                          "width": 24,
                        }
                      }
                    >
                      <i
                        className="icon"
                      >
                        <default
                          className="rightIcon"
                          height={24}
                          width={24}
                        />
                      </i>
                    </Icon>
                  </div>
                </div>
              </Input>
              This is should be a register page...
              <Button
                title="Login"
                to="login"
                type="default"
              >
                <NavLink
                  activeClassName="activeLink"
                  className="button primary"
                  to="login"
                >
                  <Link
                    aria-current={null}
                    className="button primary"
                    to="login"
                  >
                    <a
                      aria-current={null}
                      className="button primary"
                      href="/login"
                      onClick={[Function]}
                    >
                      <div
                        className="inner"
                      >
                        <p
                          className="text"
                        >
                          Login
                        </p>
                      </div>
                    </a>
                  </Link>
                </NavLink>
              </Button>
              <Button
                onClick={[Function]}
                title="Register"
                type="default"
              >
                <button
                  className="button primary"
                  onClick={[Function]}
                >
                  <p>
                    Register
                  </p>
                </button>
              </Button>
            </form>
            <Loader
              active={false}
            >
              <div
                className="loader"
              >
                <PoseGroup
                  flipMove={true}
                >
                  <Transition
                    enterAfterExit={false}
                    enterPose="enter"
                    exitPose="exit"
                    flipMove={true}
                    preEnterPose="exit"
                  />
                </PoseGroup>
              </div>
            </Loader>
          </div>
        </div>
      </Register>
    </Router>
  </BrowserRouter>
</UserProvider>
`;
